src/tribler/core/components/libtorrent/download_manager/download_manager.py

```diff
@@ -363,7 +389,7 @@ def set_max_connections(self, conns, hops=None):
     def set_upload_rate_limit(self, rate, hops=None):
         # Rate conversion due to the fact that we had a different system with Swift
         # and the old python BitTorrent core: unlimited == 0, stop == -1, else rate in kbytes
-        libtorrent_rate = int(-1 if rate == 0 else (1 if rate == -1 else rate * 1024))
+        libtorrent_rate = self.__class__.convert_rate(rate=rate)
```

Comment:
This version is a bit easier to read:

```suggestion
        libtorrent_rate = self.convert_rate(rate=rate)
```